fix(1.7.10): Overhaul Particle Accelerator

AnimeCatgirl opened the pull request:
Fixes GUI/Core bugs with the particle accelerator

This also updates mod version 1.5.5 -> 1.5.6

AnimeCatgirl committed:
fix gui bug

AnimeCatgirl committed:
fix broken hardness check

AnimeCatgirl committed:
make gui display correct amount

AnimeCatgirl committed:
make sure electromagnetic cell exists before making antimatter

AnimeCatgirl committed:
fix darkmatter dupe bug

AnimeCatgirl committed:
remove debug test that shouldnt be there

AnimeCatgirl committed:
this check breaks darkmatter creation

AnimeCatgirl committed:
update version

kmecpp commented:
I just pushed some fixes that I had forgotten to commit before which should address most or all of the PA issues. Could you retest what is still broken?

AnimeCatgirl commented:
Absolutely! I just recompiled your latest commits, and here's what still exists;
- DarkMatter dupe w/o quantum cells
- DarkMatter is not created sometimes (I disabled the random check to test correctly)
- GUI bugs
1. Mass is not shown correctly if an item is put in -> taken out -> put back in
2. Showing wrong format (although this is more of a QOL, and can be removed)
- AntiMatter is still used at 125_000(125mg) without an electromagnetic cell

AnimeCatgirl committed:
Merge branch '1.7.10' into 1.7.10

AnimeCatgirl committed:
requested changes from aurilisdev

doing it this way results in less reflection calls, causing less lag with the same desired effect
